core/src/main/java/org/jboss/jca/core/connectionmanager/pool/mcp/SemaphoreArrayListManagedConnectionPool.java

```diff
@@ -153,7 +157,34 @@ public void initialize(ManagedConnectionFactory mcf, ConnectionListenerFactory c
       this.statistics = new ManagedConnectionPoolStatisticsImpl(maxSize);
       this.statistics.setEnabled(p.getStatistics().isEnabled());
       this.permits = new Semaphore(maxSize, true, statistics);
-
+      this.filoPoolBehavior = false;
+ 
+      String value = SecurityActions.getSystemProperty("ironjacamar.filo_pool_behavior");
+ 
+      if (value != null && !value.trim().equals(""))
+      {
+         try
+         {
+            filoPoolBehavior = Boolean.valueOf(value);
+         }
+         catch (Throwable t)
+         {
+            StringTokenizer st = new StringTokenizer(value,",");
+           
+            while (st.hasMoreTokens())
+            {
+               if (st.nextToken().equals(pool.getName()))
+               {
+                  filoPoolBehavior = true;
+               }
+            }
+            if (!filoPoolBehavior)
```

Comment:
Move this 'if' outside of the system property check such that it is always invoked